### PR TITLE
fix(klaviyo): add check for listId for subscribe

### PR DESCRIPTION
**Pull request by ujjwal-ab**
## Description of the change

Add check for `listId` before creating the payload for subscribe endpoint if the subscribe flag is true(user wants to subscribe the profile)

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by utsabc — APPROVED** [review]

File: src/v0/destinations/klaviyo/transform.js [hunk 1]
```diff
@@ -99,8 +99,11 @@ const identifyRequestHandler = async (message, category, destination) => {
...
-  // check if user wants to subscribe profile or not
-  if (traitsInfo?.properties?.subscribe) {
+  // check if user wants to subscribe profile or not and listId is present or not
+  if (
+    traitsInfo?.properties?.subscribe &&
+    (traitsInfo.properties?.listId || destination.Config?.listId)
+  ) {
     responseArray.push(subscribeUserToList(message, traitsInfo, destination));
     return responseArray;
   }
```
Comment (krishna2020) [inline src/v0/destinations/klaviyo/transform.js:109]: why only first element of `responseArray` being returned ?

Comment (ujjwal-ab) [reply src/v0/destinations/klaviyo/transform.js:109]: Here, for identify call if user has sent the subscribe flag as true, then two payloads will be formed and we are returning them in an array, but if subscribe flag is false then we have only one payload and we are returning the same using reponseArray[0]

**Review by krishna2020 — APPROVED** [review]

File: src/v0/destinations/klaviyo/transform.js (hunk 1 above)
Comment (krishna2020) [reply src/v0/destinations/klaviyo/transform.js:109]: Much of the code functionality is dependent on the array length & the position of an element in array which is bad practice. 
And also, a single function having only one response type is always recommended. Here based on the scenario - either the array (of updateProfile & subscribe) or object (updateProfile) is being sent. Instead it should always return an array and have a flag in the object to determine whether it's updateProfile or subscribe
@ujjwal-ab Take this as backlog & come up with the refactoring design

Comment (ujjwal-ab) [reply src/v0/destinations/klaviyo/transform.js:109]: Will refactor it with Unit tests

**Pull request merged**